storage/Devices/FilesystemImpl.cc

```diff
@@ -378,6 +395,31 @@ namespace storage
 
 
     void
+    Filesystem::Impl::add_modify_actions(Actiongraph::Impl& actiongraph, const Device* lhs_base) const
+    {
+	const Impl& lhs = dynamic_cast<const Impl&>(lhs_base->get_impl());
+	vector<Action::Base*> actions;
+
+	if (get_type() != lhs.get_type())
```

Comment:
I don't think this can happen since a different filesystem object will have a different sid and thus the function will not be called. But the check doesn't hurt.